cloud_testing/kotlin_poc/src/main/kotlin/ftl/android/AndroidCatalog.kt

```diff
@@ -0,0 +1,36 @@
+package ftl.android
+
+import ftl.gc.GcTesting
+
+
+/**
+ * Contains lists of possible Android device and version ids, as well as checks
+ * to validate device configuration
+ */
+object AndroidCatalog {
+    private val androidDeviceCatalog by lazy {
+        GcTesting.get.testEnvironmentCatalog().get("android").execute().androidDeviceCatalog
+    }
+
+    val androidModelIds by lazy { androidDeviceCatalog.models.map { it.id } }
+    val androidVersionIds by lazy { androidDeviceCatalog.versions.map { it.id } }
+
+
+    fun supportedDeviceConfig(modelId: String, versionId: String): DeviceConfigCheck {
+        if (!androidModelIds.contains(modelId)) return UnsupportedModelId
+        if (!androidVersionIds.contains(versionId)) return UnsupportedVersionId
+
+        val supportedVersionIds = androidDeviceCatalog.models.find { it.id == modelId }?.supportedVersionIds
+        supportedVersionIds?.let {
+            if (!it.contains(versionId)) return IncompatibleModelVersion
```

Comment:
shouldn't this just throw an exception?

Reply:
exceptions are thrown when the run command is processed